5-Access-Authentication-and-Authorization/5.3-Configure-SSH-Server.bats

```diff
@@ -0,0 +1,212 @@
+    run bash -c "sshd -T -C user=root -C host=\"$(hostname)\" -C addr=\"$(grep $(hostname) /etc/hosts | awk '{print $1}')\" | grep permituserenvironment"
+    [ "$status" -eq 0 ]
+    [[ "$output" == "permituserenvironment no" ]]
+    run bash -c "grep -Eis '^\s*PermitUserEnvironment\s+yes' /etc/ssh/sshd_config /etc/ssh/sshd_config.d/*.conf"
+    [ "$status" -ne 0 ]
+    [[ "$output" == "" ]]
+}
+
+@test "5.3.13 Ensure only strong Ciphers are used (Automated)" {
+    run bash -c "sshd -T -C user=root -C host=\"$(hostname)\" -C addr=\"$(grep $(hostname) /etc/hosts | awk '{print $1}')\" | grep -Ei '^\s*ciphers\s+([^#]+,)?(3descbc|aes128-cbc|aes192-cbc|aes256-cbc|arcfour|arcfour128|arcfour256|blowfishcbc|cast128-cbc|rijndael-cbc@lysator.liu.se)\b'"
+    [ "$status" -ne 0 ]
+    [[ "$output" == "" ]]
+    run bash -c "grep -Eis '^\s*ciphers\s+([^#]+,)?(3des-cbc|aes128-cbc|aes192-cbc|aes256-cbc|arcfour|arcfour128|arcfour256|blowfish-cbc|cast128-cbc|rijndaelcbc@lysator.liu.se)\b' /etc/ssh/sshd_config /etc/ssh/sshd_config.d/*.conf"
+    [ "$status" -ne 0 ]
+    [[ "$output" == "" ]]
+}
+
+@test "5.3.14 Ensure only strong MAC algorithms are used (Automated)" {
+    run bash -c "sshd -T -C user=root -C host=\"$(hostname)\" -C addr=\"$(grep $(hostname) /etc/hosts | awk '{print $1}')\" | grep -Ei '^\s*macs\s+([^#]+,)?(hmacmd5|hmac-md5-96|hmac-ripemd160|hmac-sha1|hmac-sha1-96|umac64@openssh\.com|hmac-md5-etm@openssh\.com|hmac-md5-96-etm@openssh\.com|hmacripemd160-etm@openssh\.com|hmac-sha1-etm@openssh\.com|hmac-sha1-96-etm@openssh\.com|umac-64-etm@openssh\.com|umac-128-etm@openssh\.com)\b'"
+    [ "$status" -ne 0 ]
+    [[ "$output" == "" ]]
+    run bash -c "grep -Eis '^\s*macs\s+([^#]+,)?(hmac-md5|hmac-md5-96|hmac-ripemd160|hmacsha1|hmac-sha1-96|umac-64@openssh\.com|hmac-md5-etm@openssh\.com|hmac-md5-96-etm@openssh\.com|hmac-ripemd160-etm@openssh\.com|hmac-sha1-etm@openssh\.com|hmac-sha1-96-etm@openssh\.com|umac-64-etm@openssh\.com|umac128-etm@openssh\.com)\b' /etc/ssh/sshd_config /etc/ssh/sshd_config.d/*.conf"
+    [ "$status" -ne 0 ]
+    [[ "$output" == "" ]]
+}
+
+@test "5.3.15 Ensure only strong Key Exchange algorithms are used (Automated)" {
+    run bash -c "sshd -T -C user=root -C host=\"$(hostname)\" -C addr=\"$(grep $(hostname) /etc/hosts | awk '{print $1}')\" | grep -Ei '^\s*kexalgorithms\s+([^#]+,)?(diffie-hellman-group1-sha1|diffie-hellmangroup14-sha1|diffie-hellman-group-exchange-sha1)\b'"
+    [ "$status" -ne 0 ]
+    [[ "$output" == "" ]]
+
+    run bash -c "grep -Ei '^\s*kexalgorithms\s+([^#]+,)?(diffie-hellman-group1-sha1|diffiehellman-group14-sha1|diffie-hellman-group-exchange-sha1)\b' /etc/ssh/sshd_config"
+    [ "$status" -ne 0 ]
+    [[ "$output" == "" ]]
+}
+
+@test "5.3.16 Ensure SSH Idle Timeout Interval is configured (Automated)" {
+    local INTERVAL=$(sshd -T -C user=root -C host="$(hostname)" -C addr="$(grep $(hostname) /etc/hosts | awk '{print $1}')" | grep clientaliveinterval)
+    INTERVAL=(${INTERVAL// / })
```

Comment:
Please briefly describe what is happening in this line and all the similar ones in the following tests.
e.g. in 5.5 there was the `# get the number from the string` comment, which sounds good.
Bash is complicated, help future readers of this :D